### ISSUE TITLE
DAws.php was not found on this server.

### ISSUE THREAD

**Issue by OctAg0nO** [dotcppfile/DAws#6]
Error on running

**Comment by dotcppfile** [comment]
It's a fake error that DAws shows to trick users and bots but it's actually there. To use DAws send a post request to it holding the following post data: pass=DAws

**Comment by OctAg0nO** [comment]
after i send the post data i only get a white page.
im using hackbar to send post data.

**Comment by dotcppfile** [comment]
It should work just fine. Add me on skype maybe I can help you more (skype: dotcppfile)

**Comment by r3vO88** [comment]
Can you give me a hint how to do it? 
Simply typing "http://xxx.xxx/Daws.php?pass=DAws" dont work.

I only get : 
404 Not Found
nginx/1.2.1

**Comment by dotcppfile** [comment]
You are sending "pass=DAws" as get data while it should be sent as post data. Download a Firefox addon like Hackbar and use it to send post data easily.

**Comment by omarkurt** [comment]
POST /VulnWebApp/DAws/DAws.php HTTP/1.1
Host: debianserver
User-Agent: Mozilla/5.0 (Windows NT 6.1; WOW64; rv:40.0) Gecko/20100101 Firefox/40.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,_/_;q=0.8
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Cookie: PHPSESSID=90qbjelp3rbr9vo75jq60faca1
Connection: keep-alive

Content-Type: application/x-www-form-urlencoded
Content-Length: 9

## pass=DAws

**HTTP/1.0 500 Internal Server Error**
Date: Thu, 20 Aug 2015 16:55:26 GMT
Server: Apache/2.4.10 (Debian)
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
Pragma: no-cache
Content-Length: 0
Connection: close
Content-Type: text/html; charset=UTF-8

I seem doesn't work?

**Comment by dotcppfile** [comment]
You all just need to send pass=DAws as post data to the shell, you can easily do that threw your Firefox browser using Live HTTP Headers, Temper Data, HackBar, etc...

**Comment by webshellz** [comment]
I am getting a  Server error 500 , i've tried the newest version and the 2nd newest version and both are giving me this error. If there is any information you need to help fix this problem, let me know. An earlier version does seem to work. I was also wondering if you could make the layout look more like wso for example? and do you have a sql browser in the shell?

**Comment by dotcppfile** [comment]
Nothing in the shell should be causing an Internal Server Error and that is
why this seems like a messed up and weird issue, if you can link me to the
shell then the team and I will easily fix the issue if it exists. As for
the layout then I don't think it'll ever change; working around with HTML
and CSS is simply boring lol. As for the Sql Browser, then DAws doesn't
have one, though it does have a Sql Connector that you can obviously use to
execute sql queries.

**Comment by webshellz** [comment]
Well its not my server that I uploaded it to haha but if it helps its on
cpanel 11.52.1.3 and php version 5.4.42 , and does have
mod security enabled. Every other shell i've attempted to upload to the
server using the cpanel's file manager, gets deleted as soon as its upload.
Your shell is the only one that doesn't get deleted that i've been able to
find :P

Here's the error log it created in the folder. This is from the latest
build.

[16-Dec-2015 02:12:41 UTC] PHP Parse error:  syntax error, unexpected
'[' in /home/*/public_html/ckeditor/samples/old/assets/uilanguages/index.php
on line 212

**Comment by dotcppfile** [comment]
I'm about to push a new commit that should get your problem solved, this
wasn't really a bug just a simple syntax error caused by old versions of
PHP I guess, thanks for passing it out. Let me know if everything turns out
to be good.

**Comment by webshellz** [comment]
missed one , syntax error, unexpected '[' on line 1880

**Comment by dotcppfile** [comment]
Are you sure about that? Because there's nothing on line 1880 except for HTML.

**Comment by webshellz** [comment]
```
if (installed_php("posix_getpwuid")) {
                        $fileowner = posix_getpwuid($fileowner)["name"];
                        $filegroup = posix_getgrgid($filegroup)["name"];
                    }
```

this was the part that was giving me the syntax error but i fixed it for myself and i got it to work! But i was wondering if there was a way you could change the folder it created, trying to make it as stealthy as possible... possibly a common folder name?

**Comment by dotcppfile** [comment]
I didn't check that part and I think you're right so I pushed a new commit
for this, thanks for reporting mate.
The folder name is "DAws", changing it to something common requires
necessary checks, so what you're requesting isn't something simple but I'll
take it in consideration even though I think that DAws shouldn't be
considered as a permanent backdoor but more like a big gun that you drop to
get the job done.

But if you incest then here's what you need to do for your "private"
version of DAws:
Update lines 448 and 449 by changing the "DAws" values to whatever you want
and that should be it.

Good luck,
dotcppfile.

**Comment by EJ207** [comment]
I am getting this same issue on the latest version of DAws .

I have tried it on multiple servers as well as XAMPP locally. 

Every time: "The requested URL /DAws/DAws.php was not found on this server."

My XAMPP config: Apache/2.4.17 (Win32) OpenSSL/1.0.2d PHP/5.6.23 at ::1 Port 80

Or it tries to redirect into the parent directory or something and provides this error on one server (I run them all, this is for educational and training purposes). So im very excited to try it out, my next resort is to review the code myself =\ 

"The server can not find the requested page:

xx.xx.123.134/DAws.php (port 80)"

So im very excited to try it out, my next resort is to review the code myself =\

**Comment by awnumar** [comment]
@ej207 Did you pass DAws as a POST parameter?

**Comment by EJ207** [comment]
I Just did actually, This should be specified in plain english in the README or somewhere more clearly. I did not find this information except for in this git issue here. Thanks for your response.
Edit: to clarify, after passing the pass through POST headers it is working.

**Comment by EJ207** [comment]
I do get the error using a production web server though, I will keep trying other methods of passing the headers maybe that is the issue in this case but it works great on xampp apache/php
